api-module-library/hubspot/tests/api.test.js

```diff
@@ -1,26 +1,376 @@
-const Manager = require('./manager');
-const mongoose = require('mongoose');
-const config = require('./defaultConfig.json');
+const {Api} = require('../api');
+const config = require('../defaultConfig.json');
+const Authenticator = require('@friggframework/test-environment/Authenticator');
 
-describe(`Should fully test the ${config.label} Manager`, () => {
-    let manager, userManager;
+describe(`${config.label} API tests`, () => {
+
+
+    const apiParams = {
+        client_id: process.env.HUBSPOT_CLIENT_ID,
+        client_secret: process.env.HUBSPOT_CLIENT_SECRET,
+        redirect_uri: `${process.env.REDIRECT_URI}/hubspot`,
+        scope: process.env.HUBSPOT_SCOPE
+    };
+    const api = new Api(apiParams);
 
     beforeAll(async () => {
```

Comment:
Medium: except if necessary (it is unclear if these tests are mocking the API calls at all) I'd prefer to use `beforeEach` instead.

As `beforeAll` is invoked only once for the entire block it belongs, it can result in stateful tests if any of the parts consuming the values associated here ever touch any value - a condition that is very hard to trace.

Reply:
marking as resolved because this occurs in the "legacy" code and is noted in it's need for refactor
(discussion on this topic moved to slack)